### PR TITLE
Fix skeleton.toDB.txnUpdate

### LINKED ISSUE #110: Exception raised in several bones when working with sub-skeletons
I have a skeleton with subSkels:
```python
class optionSkel(skeleton.Skeleton):

	subSkels = {
		"*": [
			"kind",
			"kind_code",
			"code",
			"creationdate",
			"changedate"
		],
		"height": [
		],
		"color": [
			"name",
			"image"
		]
	}

	kind = selectBone(
		descr=u"Art",
		indexed=True,
		visible=False,
		readOnly=True,
		required=True,
		values=["height", "color"]
	)

	kind_code = stringBone(
		descr=u"Zusammengesetzter Code",
		visible=False,
		required=True,
		unique=True,
		indexed=True
	)

	code = codeBone(
		descr=u"Code",
		invalidChars="*", #The star "*" means "empty/unset" in the resulting variant_id.
		required=True,
		indexed=True
	)

	name = stringBone(
		descr=u"Bezeichnung",
		required=True,
		searchable=True,
		indexed=True
	)

	image = fileBone(
		descr=u"Bild"
	)
```
Now I got an skeleton instance using
```python
skel = optionSkel.subSkel("height")
skel.fromDB(...)
skel["name"] = "Bla"
skel.toDB()
```
and it fails with this error:
```
Traceback (most recent call last):
  File "/home/neo/MyProject/deploy/server/tasks.py", line 147, in deferred
    caller(*args, **kwargs)
  File "/home/neo/MyProject/deploy/server/tasks.py", line 316, in <lambda>
    return( lambda *args, **kwargs: mkDefered( func, *args, **kwargs) )
  File "/home/neo/MyProject/deploy/server/tasks.py", line 278, in mkDefered
    return func(self, *args, **kwargs)
  File "/home/neo/MyProject/deploy/prototypes/importable.py", line 272, in doPrepareImportdate
    assert skel.toDB(clearUpdateTag=True)
  File "/home/neo/MyProject/deploy/skeletons/option.py", line 165, in toDB
    return super(optionSkel, self).toDB(*args, **kwargs)
  File "/home/neo/MyProject/deploy/server/skeleton.py", line 729, in toDB
    txnUpdate, key, self, clearUpdateTag)
  File "/home/neo/bin/google-cloud-sdk/platform/google_appengine/google/appengine/api/datastore.py", line 2635, in RunInTransactionOptions
    function, *args, **kwargs)
  File "/home/neo/bin/google-cloud-sdk/platform/google_appengine/google/appengine/api/datastore.py", line 2710, in _RunInTransactionInternal
    ok, result = _DoOneTry(function, args, kwargs)
  File "/home/neo/bin/google-cloud-sdk/platform/google_appengine/google/appengine/api/datastore.py", line 2748, in _DoOneTry
    result = function(*args, **kwargs)
  File "/home/neo/MyProject/deploy/server/skeleton.py", line 602, in txnUpdate
    blobList.update(_bone.getReferencedBlobs(self.valuesCache, key))
  File "/home/neo/MyProject/deploy/server/bones/fileBone.py", line 19, in getReferencedBlobs
    if valuesCache[name] is None:
KeyError: 'image'
```
In my opinion, the loops at line 592 in server/skeleton.py reading
```python
## Merge the values from mergeFrom in
for key, bone in skel.items():
	if key in mergeFrom:
		bone.mergeFrom(skel.valuesCache, key, mergeFrom)
for key, _bone in skel.items():
	dbObj = _bone.serialize(skel.valuesCache, key, dbObj)
	blobList.update(_bone.getReferencedBlobs(self.valuesCache, key))
```
should be merged into one loop and changed into this:
```python
# Merge the values from mergeFrom in
for key, bone in skel.items():
	if key in mergeFrom:
		bone.mergeFrom(skel.valuesCache, key, mergeFrom)

	if key in self.valuesCache:
		dbObj = bone.serialize(skel.valuesCache, key, dbObj)
		blobList.update(bone.getReferencedBlobs(self.valuesCache, key))
```
Am I right?

### PR DESCRIPTION
**Pull request by phorward**
This is a pull-request as proposal to resolve #110 and reduce the number of loops inside the function.

### CODE REVIEW

**Comment by phorward** [comment]
Hi, I've pushed a corrected version that calls bone.serialize() always now.
I did a test run with this skeleton.py in a huger project without any problems.

File: skeleton.py [hunk 1]
```diff
@@ -564,124 +564,155 @@ def toDB(self, clearUpdateTag=False):
...
 					bone.mergeFrom(skel.valuesCache, key, mergeFrom)
-			for key, _bone in skel.items():
-				dbObj = _bone.serialize(skel.valuesCache, key, dbObj)
-				blobList.update(_bone.getReferencedBlobs(self.valuesCache, key))
+
+				# Serialize bone into entity
+				dbObj = bone.serialize(skel.valuesCache, key, dbObj)
+
+				# Obtain referenced blobs
+				if key in self.valuesCache:
```
Comment (tsteinruecken) [inline skeleton.py]: This is not identical to the old logic. Why should we make assumptions about the layout inside the entity here? We simple can call getReferencedBlobs and let the bone sort things out

File: skeleton.py [hunk 2]
```diff
@@ -564,124 +564,155 @@ def toDB(self, clearUpdateTag=False):
...
+
 					else:
-						if "%s.uniqueIndexValue" % boneName in dbObj:
-							del dbObj["%s.uniqueIndexValue" % boneName]
+						if "%s.uniqueIndexValue" % key in dbObj:
+							del dbObj["%s.uniqueIndexValue" % key]
+
+				if not skel.searchIndex:
+					# We generate the search index using the full skel, not this (maybe incomplete one)
+					if bone.searchable and key in self.valuesCache:
```
Comment (tsteinruecken) [inline skeleton.py]: Same thing - we shouldn't make any assumptions here...

File: skeleton.py (hunk 1 above)
Comment (phorward) [reply skeleton.py]: Do you mean that `bone.getReferencedBlobs()` should check for its own bone existence?
This is OK for me, but I know that this was the causing for several issues in the past, when a bone does not check for its keys existence within the current cache.

Comment (phorward) [reply skeleton.py]: I've fixed it now.

**Pull request merged**